compio-driver/src/key.rs

```diff
@@ -189,6 +195,19 @@ impl<T> Key<T> {
         let op = unsafe { Box::from_raw(self.user_data as *mut RawOp<T>) };
         BufResult(op.result.take_ready().unwrap_unchecked(), op.op)
     }
+
+    /// Get the inner result and flags if it is completed.
+    ///
+    /// # Safety
+    ///
+    /// Call it only when the op is completed, otherwise it is UB.
+    pub(crate) unsafe fn into_inner_flags(self) -> (BufResult<usize, T>, u32) {
```

Comment:
Just a method `flags(&self) -> u32` is fine, I think. This method seems redundant.